inc/functions/cloudflare.php

```diff
@@ -41,15 +41,44 @@ function rocket_is_api_keys_valid_cloudflare( $cf_email, $cf_api_key, $cf_zone_i
 
 	try {
 		$cf_api_instance = new Cloudflare\Api( $cf_email, $cf_api_key );
-		$cf_user         = $cf_api_instance->get('user/');
-		$cf_zone         = $cf_api_instance->get('zones/'.$cf_zone_id);
+		$cf_user         = $cf_api_instance->get( 'user/' );
+		$cf_zone         = $cf_api_instance->get( 'zones/' . $cf_zone_id );
 
 		if ( ! isset( $cf_zone->success ) || empty( $cf_zone->success ) ) {
 			return new WP_Error( 'cloudflare_invalid_auth', $cf_zone->errors[0]->message );
 		}
 
 		if ( true === $cf_zone->success ) {
-		    return true;
+			$zone_found = false;
+			$site_url   = get_site_url();
+
+			if ( function_exists( 'domain_mapping_siteurl' ) ) {
+				$site_url = domain_mapping_siteurl( $site_url );
+			}
+
+			if ( isset( $cf_zone->result ) && ! empty( $cf_zone->result ) ) {
+				foreach ( $cf_zone->result as $zone ) {
+					if ( strtolower( $site_url ) === $zone->name ) {
```

Comment:
This `===` feels too simple to break:
- URL schemes may differ (http vs https), or be absent.
- Trailing slash wants you dead.

A small normalization helper may help:

```php
$url = strtolower( $url );
$url = preg_replace( '@^https?:@', '', $url );
$url = trim( $url, '/' );
```

Reply:
What do you think about using `parse_url` and compare only the host?
`parse_url('http://abc.net/', PHP_URL_HOST);`

`$site_url` - is Wordpress URL
`$zone->name` - is coming as www.abc.com or abc.com

Reply:
If the host is enough, yes, no problem.
(sorry didn't see your answer)